distribution/tools/keystore-cli/src/test/java/org/elasticsearch/common/settings/AddFileKeyStoreCommandTests.java

```diff
@@ -53,110 +53,156 @@ private Path createRandomFile() throws IOException {
-        addFile(keystore, "foo", file);
+        KeyStoreWrapper keystore = createKeystore(password);
+        addFile(keystore, "foo", file, password);
+        terminal.addSecretInput(password);
+        terminal.addSecretInput(password);
         terminal.addTextInput("");
         execute("foo", "path/dne");
-        assertSecureFile("foo", file);
+        assertSecureFile("foo", file, password);
     }
 
     public void testOverwritePromptExplicitNo() throws Exception {
+        String password = "keystorepassword";
         Path file = createRandomFile();
-        KeyStoreWrapper keystore = createKeystore("");
-        addFile(keystore, "foo", file);
+        KeyStoreWrapper keystore = createKeystore(password);
+        addFile(keystore, "foo", file, password);
+        terminal.addSecretInput(password);
         terminal.addTextInput("n"); // explicit no
         execute("foo", "path/dne");
-        assertSecureFile("foo", file);
+        assertSecureFile("foo", file, password);
     }
 
     public void testOverwritePromptExplicitYes() throws Exception {
+        String password = "keystorepassword";
         Path file1 = createRandomFile();
-        KeyStoreWrapper keystore = createKeystore("");
-        addFile(keystore, "foo", file1);
+        KeyStoreWrapper keystore = createKeystore(password);
+        addFile(keystore, "foo", file1, password);
+        terminal.addSecretInput(password);
+        terminal.addSecretInput(password);
         terminal.addTextInput("y");
         Path file2 = createRandomFile();
         execute("foo", file2.toString());
-        assertSecureFile("foo", file2);
+        assertSecureFile("foo", file2, password);
     }
 
     public void testOverwriteForceShort() throws Exception {
+        String password = "keystorepassword";
         Path file1 = createRandomFile();
-        KeyStoreWrapper keystore = createKeystore("");
-        addFile(keystore, "foo", file1);
+        KeyStoreWrapper keystore = createKeystore(password);
+        addFile(keystore, "foo", file1, password);
         Path file2 = createRandomFile();
+        terminal.addSecretInput(password);
+        terminal.addSecretInput(password);
         execute("-f", "foo", file2.toString());
-        assertSecureFile("foo", file2);
+        assertSecureFile("foo", file2, password);
     }
 
     public void testOverwriteForceLong() throws Exception {
+        String password = "keystorepassword";
         Path file1 = createRandomFile();
-        KeyStoreWrapper keystore = createKeystore("");
-        addFile(keystore, "foo", file1);
+        KeyStoreWrapper keystore = createKeystore(password);
+        addFile(keystore, "foo", file1, password);
         Path file2 = createRandomFile();
+        terminal.addSecretInput(password);
         execute("--force", "foo", file2.toString());
-        assertSecureFile("foo", file2);
+        assertSecureFile("foo", file2, password);
     }
 
     public void testForceNonExistent() throws Exception {
```

Comment:
Nit-pick - I first read this test name as meaning that the file being added doesn't exist. Could we tweak it, e.g.

```suggestion
    public void testForceDoesNotAlreadyExist() throws Exception {
```

More generally, it'd be great if these tests had Javadoc comments that briefly explained each scenario / expectation, but I'll leave that up to you.

Reply:
I think this is a generic problem for a lot of our tests. It's not quite obvious all the time what the test does and the method name can only have so many words in it. However, very few of our tests have javadocs. 

In this case I think we can get away with editing the method names but feel free to argue against it